fix: Remove bit-fields

JonathanWoollett-Light opened the pull request:
## Changes

Removes `bit-fields` and `bit-fields-macros` crates.

## Reason

They are not used frequently enough to warrant the additional complexity.

## License Acceptance

By submitting this pull request, I confirm that my contribution is made under
the terms of the Apache 2.0 license. For more information on following
Developer Certificate of Origin and signing off your commits, please check
[`CONTRIBUTING.md`][3].

## PR Checklist

- [x] If a specific issue led to this PR, this PR closes the issue.
- [x] The description of changes is clear and encompassing.
- [x] Any required documentation changes (code and docs) are included in this PR.
- [x] API changes follow the [Runbook for Firecracker API changes][2].
- [x] User-facing changes are mentioned in `CHANGELOG.md`.
- [x] All added/changed functionality is tested.
- [x] New `TODO`s link to an issue.
- [x] Commits meet [contribution quality standards](https://github.com/firecracker-microvm/firecracker/blob/main/CONTRIBUTING.md#contribution-quality-standards).

---

- [x] This functionality cannot be added in [`rust-vmm`][1].

[1]: https://github.com/rust-vmm
[2]: https://github.com/firecracker-microvm/firecracker/blob/main/docs/api-change-runbook.md
[3]: https://github.com/firecracker-microvm/firecracker/blob/main/CONTRIBUTING.md

ShadowCurse commented on src/vmm/src/guest_config/cpuid/normalize.rs:
```diff
@@ -50,6 +49,80 @@ pub enum GetMaxCpusPerPackageError {
+    clippy::arithmetic_side_effects,
+    clippy::unreachable
+)]
+const fn shift(x: u8, max: u8) -> u32 {
```
Shouldn't here be another `debug_assert!(max < 32)`? Otherwise you can call it with `shift(100, 200)`, but it does not make sense.
Or maybe just add it as a lambda inside `mask` method?

ShadowCurse replied:
nit: Maybe a little refactoring might be useful. Like removing `unreachable` + `shift` technically does not need second arg, because it is always 32.

ShadowCurse replied:
Ok, this version is looks good to me too.

bchalios reviewed (commented):
Could we please create PRs and commit messages that are a bit more informative on what we are doing and why?

JonathanWoollett-Light commented:
> Could we please create PRs and commit messages that are a bit more informative on what we are doing and why?

Added commit body (and added this to PR description).
